socketshark/__init__.py

```diff
@@ -126,35 +127,32 @@ async def prepare(self):
 
         Initialize Redis connection and the receiver class.
         """
-        redis_receiver = Receiver(loop=asyncio.get_event_loop())
-        redis_settings = self.config['REDIS']
+        redis_settings = [self.config['REDIS']]
+        if self.config.get('REDIS_ALT'):
+            redis_settings.append(self.config['REDIS_ALT'])
+        self.redis_connections = []
         try:
-            self.redis = await aioredis.create_redis(
-                (redis_settings['host'], redis_settings['port']),
-                db=redis_settings.get('db', 0),
-            )
+            for settings in redis_settings:
+                self.redis_connections.append(
+                    await RedisConnection.create(settings)
+                )
         except (OSError, aioredis.RedisError):
             self.log.exception('could not connect to redis')
             raise
-
-        # Some features (e.g. pinging) don't work on old Redis versions.
-        info = await self.redis.info('server')
-        version_info = info['server']['redis_version'].split('.')
-        major, minor = int(version_info[0]), int(version_info[1])
-        if not (major > 3 or major == 3 and minor >= 2):
-            msg = 'Redis version must be at least 3.2'
-            self.log.exception(msg, version_info=version_info)
-            raise Exception(msg)
+        except Exception:
+            self.log.exception('unexpected error')
+            raise
```

Comment:
Do we need this?

Reply:
This catches an exception thrown by RedisConnection, covering the version check. I want to log here for the context. I will convert this into a specific exception for clarity.

Reply:
I added a specific exception in [318608f](https://github.com/closeio/socketshark/pull/105/commits/318608faf0f80511ec41098119f746b2a0257090)

Reply:
Right, but what's the difference between a) not catching an exception, b) catching an exception, logging it as an exception, and then re-raising the exception? If we did (a) and never caught the exception, wouldn't it already be logged and there would be no need for re-raising?

Reply:
Yes, fair point. I removed this catch/raise here.